lib/ddtrace/contrib/sidekiq/client_tracer.rb

```diff
@@ -0,0 +1,28 @@
+require 'ddtrace/contrib/sidekiq/tracing'
+
+module Datadog
+  module Contrib
+    module Sidekiq
+      # Tracer is a Sidekiq client-side middleware which traces job enqueues/pushes
+      class ClientTracer
+        include Tracing
+
+        # Client middleware arguments are documented here:
+        #   https://github.com/mperham/sidekiq/wiki/Middleware#client-middleware
+        def call(worker_class, job, queue, redis_pool)
+          resource = job_resource(job)
+          service = sidekiq_service(resource)
```

Comment:
Just following up on what service names should be, we should 1) avoid having the service match the resource, and 2) the server/client service names matching each other.

For number 1, the service name should act as a group name for a logical deployment of Sidekiq (e.g. application), as users will want to see application performance statistics aggregated by role first, then broken down by task (which is where resource comes into play.) Users should be able to configure the client and service names separately from one another, so to support that, we might need to add in another configuration option like `client_service_name` or something similar.

For number 2, the client shouldn't match the server because otherwise we see some conflation of statistics between the two. We should probably keep the default service name for the server side as `sidekiq` (so as to not break user's dependency on this existing default), and give the client service a new default, perhaps something like `sidekiq-client` or similar.

Reply:
Oh, okay, I may have misread the implementation here a bit. I see you're not copying the resource in for the service name, but mapping it to some other service name that might exist on the class in question. This starts to cross into what we've referred to as "multiplexing"; mapping tracer configuration based on trace content.

Is this important to implement in this PR? (The mapping resource --> service name on the job?) I might suggest keeping it simple with straight configuration variables from `Datadog.configuration[:sidekiq]` for now, since this is a bit more complicated, and we have a new prescribed way of doing such mappings as a part of the integration's configuration (such as what was done in #451.)

If we keep this simple in this PR, we can probably get this PR merged and open another one for the multiplexing feature. Might be the easiest way to get parts of this out more quickly.